client/state/utils.js

```diff
@@ -19,3 +23,34 @@ export function isValidStateWithSchema( state, schema, checkForCycles = false, b
 	}
 	return result.valid;
 }
+
+export function createReducer( initialState = null, handlers = {}, schema = null ) {
+	const defaultHandlers = {
+		[SERIALIZE]: ( state ) => {
+			if ( schema !== null ) {
+				return state;
+			}
+
+			return initialState;
+		},
+		[DESERIALIZE]: ( state ) => {
+			if ( schema !== null && isValidStateWithSchema( state, schema ) ) {
+				return state;
+			}
+
+			return initialState;
+		}
+	};
+
+	handlers = Object.assign( {}, defaultHandlers, handlers );
```

Comment:
We could optimize this a bit to avoid setting up default handlers (and, more importantly, running on every `SERIALIZE` and `DESERIALIZE` when no schema is defined), by rewriting this:

```js
if ( schema ) {
	handlers = Object.assign( {
		[ SERIALIZE ]: ( state ) => {
			return initialState;
		},
		[ DESERIALIZE ]: ( state ) => {
			if ( isValidStateWithSchema( state, schema ) ) {
				return state;
			}

			return initialState;
		}
	}, handlers );
}
```